src/gamma_inc.jl

```diff
@@ -943,33 +954,40 @@ function _gamma_inc_inv(a::Float64, p::Float64, q::Float64)
         x0 = gamma_inc_inv_asmall(a, p, q, pcase)
     else    #large a
         haseta = true
-        (x0,fp) = gamma_inc_inv_alarge(a, porq, s)
+        x0, fp = gamma_inc_inv_alarge(a, porq, s)
     end
 
     t = 1
     x = x0
     n = 1
     logabsgam = logabsgamma(a)[1]
-    #Newton-like higher order iteration with upper limit as 15.
+    # Newton-like higher order iteration with upper limit as 15.
     while t > 1.0e-15 && n < 15
         x = x0
         x² = x*x
         if !haseta
             dlnr = (1.0 - a)*log(x) + x + logabsgam
             if dlnr > log(floatmax(Float64)/1000.0)
-                n = 20
+                break
             else
                 r = exp(dlnr)
             end
         else
             r = -(1/fp)*x
         end
 
-        (px, qx) = gamma_inc(a, x, 0)
-        ck1 = pcase ? -r*(px-p) : r*(qx-q)
+        px, qx = gamma_inc(a, x, 0)
+        ck1 = pcase ? -r*(px - p) : r*(qx - q)
         ck2 = (x - a + 1.0)/(2.0*x)
         ck3 = (@horner(x, @horner(a, 1, -3, 2), @horner(a, 4, -4), 2))/(6*x²)
         r = ck1
+
+        # This check is not in the invincgam subroutine from IncgamFI but it probably
+        # should be since the routine fails to compute a finite value for very small p
+        if !isfinite(ck3)
+            break
+        end
```

Comment:
I guess the check `t > 1e-15` would terminate the loop anyway but at this point `x` is already set to a non-finite value? So it seems reasonable to include this check.

However, more generally, it seems `ck3` is only needed if `a > 0.1` and `ck2` only if `a > 0.05` (and `a > 0.1`), so maybe the computations of these two terms and also the check `isfinite(ck3)` could be moved to the corresponding branches below?

Reply:
Yes. That makes sense. Then I'd have to modify the check a bit since I think such a check is needed also in the case where only ch2 is required. I'll try it out. The new tests should reveal any issues.

Reply:
Regarding your first point then the issue is that for small `x` then `x²` is zero and `ck3` becomes `Inf` which again causes `NaN`s in the polynomial evaluations. I've updated the PR according to your suggestions.